addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -540,15 +540,16 @@ const Wysiwyg = Widget.extend({
             }
             if (forceOpen || !this.linkTools) {
                 const $btn = this.toolbar.$el.find('#create-link');
-                this.linkTools = new weWidgets.LinkTools(this, {wysiwyg: this}, this.odooEditor.editable, $btn, link);
+                this.linkTools = new weWidgets.LinkTools(this, {wysiwyg: this}, this.odooEditor.editable, {}, $btn, link);
                 this.linkTools.appendTo(this.toolbar.$el);
             } else {
                 this.linkTools = undefined;
             }
         } else {
             const linkDialog = new weWidgets.LinkDialog(this, {
                 forceNewWindow: this.options.linkForceNewWindow,
-            }, this.$editable[0], undefined, link);
+                wysiwyg: this,
+            }, this.$editable[0], {}, undefined, link);
             linkDialog.open();
             linkDialog.on('save', this, (linkInfo) => {
                 const linkUrl = linkInfo.url;
```

Comment:
I would merge/reuse the code to insert the link that linktool is using.

Reply:
Done.